### PR TITLE
Cascader: optimize hover to expand experience

### PR DESCRIPTION
**Pull request by Dzinlife**
group

When cursor moves between menus, the submenus switch immediately, it is hard to select the desired item.

A better way is to include a bit of a delay when activating submenus, but it will make the menu switch slowly.

The perfect UX is to keep menus switch fast, only hold it when we want to select the submenus. 
Here is a solution:

group

Idea from OSX. The red area prevent menus switch for a short time.

### CODE REVIEW

File: packages/cascader/src/menu.vue [hunk 1]
```diff
@@ -138,6 +138,33 @@
         popperClass
       } = this;
 
+      let hoverMenuRefs = {};
+      this.hoverTimer = null;
```
Comment (Leopoldthecoder) [inline packages/cascader/src/menu.vue]: `this.hoverTimer` 最好在 data 里注册一下

File: packages/cascader/src/menu.vue [hunk 2]
```diff
@@ -138,6 +138,33 @@
...
+      const hoverMenuHandler = e => {
+        const hoverHreshold = 500;
```
Comment (Leopoldthecoder) [inline packages/cascader/src/menu.vue]: `hoverThreshold`

**Comment by Leopoldthecoder** [comment]
最好自己 fork 一下项目，从自己项目的分支提 PR，这样能避免在主仓库直接创建分支。

**Comment by Dzinlife** [comment]
嗯嗯好的

**Pull request merged**